pkg/provider/aws/knowledgebase/api_gateway.go

```diff
@@ -105,9 +126,10 @@ var ApiGatewayKB = knowledgebase.Build(
 				tg.TargetType = "instance"
 				dag.AddDependency(tg, instance)
 			}
+			tg.Port = 3000
 
-			if !isEc2Dest && !isTgDest {
-				return fmt.Errorf("destination of api integration -> load balancer expansion must be a target group or ec2 instance, but got %T", data.Destination)
+			if !isEc2Dest && !isTgDest && !isEcsDest {
+				return fmt.Errorf("destination of api integration -> load balancer expansion must be a target group, ecs service, or ec2 instance, but got %T", data.Destination)
```

Comment:
This implementation is definitely more focused on the idea of 1 service  -> 1 container -> 1 target group rather than 1 service -> any number of containers -> any number of target groups.

A target group gets created for a service if an `ApiIntegration` -> `EcsService` edge exists and then it gets mapped to a container/port in the service's task when configuring the `EcsService` -> `TargetGroup` edge.